app/dash_app.py

```diff
@@ -126,44 +132,91 @@ def _read_coherence(coherence_csv):
+                        'btemp',
+                        'bperp2',
+                        'x']
+    baseline = baseline.drop(['index',
+                              'btemp',
+                              'bperp2',
+                              'x'], axis=1)
+    baseline['first_date'] = pd.to_datetime(baseline['first_date'],
+                                            format="%Y%m%d")
+    baseline['second_date'] = pd.to_datetime(baseline['second_date'],
+                                             format="%Y%m%d")
+    return baseline
+
+
 def _valid_dates(coh):
     return coh.first_date.dropna().unique()
 
 
 def _coherence_csv(target_id):
-    site, beam = target_id.split('_', 1)
+    site, beam = target_id.rsplit('_', 1)
     return f'Data/{site}/{beam}/CoherenceMatrix.csv'
 
 
+def _baseline_csv(target_id):
+    site, beam = target_id.rsplit('_', 1)
+    return f'Data/{site}/{beam}/bperp_all'
+
+
 def pivot_and_clean(coh_long):
     """Convert long-form coherence to wide-form and clean it up."""
     coh_wide = coh_long.pivot(
         index='delta_days',
         columns='second_date',
         values='coherence')
-
     # include zero baseline even though it will never be valid
     coh_wide.loc[0, :] = np.NaN
     coh_wide.sort_index(inplace=True)
-
     # because hovertemplate 'f' format doesn't handle NaN properly
     coh_wide = coh_wide.round(2)
-
     # trim empty edges
     coh_wide = coh_wide.loc[
         (coh_wide.index >= 0) &
         (coh_wide.index <= coh_wide.max(axis='columns').last_valid_index()),
         (coh_wide.columns >= coh_wide.max(axis='index').first_valid_index()) &
         (coh_wide.columns <= coh_wide.max(axis='index').last_valid_index())]
-
     return coh_wide
 
 
+def pivot_and_clean_dates(coh_long, coh_wide):
+    """Convert long-form df to wide-form date matrix matching coh_wide."""
+    coh_long = coh_long.drop(coh_long[coh_long.second_date <
+                                      coh_long.first_date].index)
+    date_wide = coh_long.pivot(
+        index='delta_days',
+        columns='second_date',
+        values='first_date')
+    date_wide = date_wide.applymap(lambda x: pd.to_datetime(x)
+                                   .strftime('%b %d, %Y') if x is not pd.NaT
+                                   else x)
+    # remove the first row so that date_wide has
+    # the same columns as coh_wide
+    common_cols = [col for col in
+                   set(date_wide.columns).intersection(coh_wide.columns)]
+    common_cols.sort()
+    date_wide = date_wide[common_cols]
+    # date_wide.drop(columns=date_wide.columns[0], axis=1,  inplace=True)
+    return date_wide
+
+
 def plot_coherence(coh_long):
     """Plot coherence for different baselines as a function of time."""
     coh_long['delta_days'] = (coh_long.second_date -
                               coh_long.first_date).dt.days
     coh_wide = pivot_and_clean(coh_long)
+    date_wide = pivot_and_clean_dates(coh_long, coh_wide)
```

Comment:
Actually, I undersand now. I think you should just put this cleaning of dates into "pivot_and_clean()". That was the intent.

Reply:
This could be done, but it would mean one function returning two separate outputs. Would also make the combined function excessively long. Suggest we backlog and revisit if actually necessary

Reply:
Totally ok for a method to return two variables!

Reply:
This would contradict clean code principles. A good reference is Robert C. Martin's book by the same name. Here's a short summary of the relevant principles for functions:

> How to Write Functions
> Keep them Small
> Functions should be small, really small. They should rarely be 20 lines long. The longer a function gets, it is more likely it is to do multiple things and have side effects.
> 
> Make Sure They Just Do One Thing
> Functions should do one thing. They should do it well. They should do it only. – Clean Code
> Your functions should do only one thing. If you follow this rule, it is guaranteed that they will be small. The only thing that function does should be stated in its name.
> 
> Sometimes it is hard to look at the function and see if it is doing multiple things or not. One good way to check is to try to extract another function with a different name. If you can find it, that means it should be a different function.
> 
> This is probably the most important concept in this article, and it will take some time to get used to. But once you get the hang of it, your code will look much more mature, and it will be more easily refactorable, understandable, and testable for sure. (https://www.freecodecamp.org/news/clean-coding-for-beginners/)